Mathlib/MeasureTheory/Function/Holder.lean

```diff
@@ -0,0 +1,255 @@
+  simp only [holder, ← Memℒp.toLp_const_smul]
+  apply Memℒp.toLp_congr
+  filter_upwards [Lp.coeFn_smul c g] with x hx
+  simp [hx]
+
+variable (μ p q r) in
+/-- `MeasureTheory.Lp.holder` as a bilinear map. -/
+@[simps! apply_apply]
+def holderₗ : Lp E p μ →ₗ[𝕜] Lp F q μ →ₗ[𝕜] Lp G r μ :=
+  .mk₂ 𝕜 B.holder B.holder_add_left B.holder_smul_left
+    B.holder_add_right B.holder_smul_right
+
+variable [Fact (1 ≤ p)] [Fact (1 ≤ q)] [Fact (1 ≤ r)]
+
+variable (μ p q r) in
+/-- `MeasureTheory.Lp.holder` as a continuous bilinear map. -/
+@[simps! apply_apply]
+def holderL : Lp E p μ →L[𝕜] Lp F q μ →L[𝕜] Lp G r μ :=
+  LinearMap.mkContinuous₂ (B.holderₗ μ p q r) ‖B‖ (norm_holder_apply_apply_le B)
+
+lemma norm_holderL_le : ‖(B.holderL μ p q r)‖ ≤ ‖B‖ :=
+  LinearMap.mkContinuous₂_norm_le _ (norm_nonneg B) _
+
+variable [HolderConjugate p q] [NormedSpace ℝ G] [SMulCommClass ℝ 𝕜 G] [CompleteSpace G]
+
+variable (μ p q) in
+/-- The natural pairing between `Lp E p μ` and `Lp F q μ` (for Hölder conjugate `p q : ℝ≥0∞`) with
+values in a space `G` induced by a bilinear map `B : E →L[𝕜] F →L[𝕜] G`.
+
+This is given by `∫ x, B (f x) (g x) ∂μ`.
+
+In the special case when `B := (NormedSpace.inclusionInDoubleDual 𝕜 E).flip`, which is
+definitionally the same as `B := ContinuousLinearMap.id 𝕜 (E →L[𝕜] 𝕜)`, this is the
+natural map `Lp (Dual 𝕜 E) p μ →L[𝕜] Dual 𝕜 (Lp E q μ)`. -/
+def lpPairing (B : E →L[𝕜] F →L[𝕜] G) : Lp E p μ →L[𝕜] Lp F q μ →L[𝕜] G :=
+  (L1.integralCLM' 𝕜 |>.postcomp <| Lp F q μ) ∘L (B.holderL μ p q 1)
+
+lemma lpPairing_eq_integral (f : Lp E p μ) (g : Lp F q μ) :
+    B.lpPairing μ p q f g = ∫ x, B (f x) (g x) ∂μ := by
+  show L1.integralCLM _ = _
+  rw [← L1.integral_def, L1.integral_eq_integral]
+  exact integral_congr_ae <| B.coeFn_holder _ _
+
+end ContinuousLinearMap
+
+end Bilinear
+
+namespace MeasureTheory
+namespace Lp
+
+/-! ### Heterogeneous scalar multiplication
+
+While the previous section is *nominally* more general than this one, and indeed, we could
+use the constructions of the previous section to define the scalar multiplication herein,
+we would lose some slight generality as we would need to require that `𝕜` is a nontrivially
+normed field everywhere. Moreover, it would only simplify a few proofs.
+-/
+
+section SMul
+
+variable {α 𝕜' 𝕜 E : Type*} {m : MeasurableSpace α} {μ : Measure α}
+    {p q r : ℝ≥0∞} [hpqr : HolderTriple p q r]
+
+section MulActionWithZero
+
+variable [NormedRing 𝕜] [NormedAddCommGroup E] [MulActionWithZero 𝕜 E] [BoundedSMul 𝕜 E]
+
+/-- Heterogeneous scalar multiplication of `MeasureTheory.Lp` functions by `MeasureTheory.Lp`
+functions when the exponents satisfy `ENNReal.HolderTriple p q r`. -/
+instance : HSMul (Lp 𝕜 p μ) (Lp E q μ) (Lp E r μ) where
+  hSMul f g := (Lp.memℒp g).smul (Lp.memℒp f) |>.toLp (⇑f • ⇑g)
+
+lemma smul_def {f : Lp 𝕜 p μ} {g : Lp E q μ} :
+    f • g = ((Lp.memℒp g).smul (Lp.memℒp f)).toLp (⇑f • ⇑g) :=
+  rfl
+
+lemma coeFn_lp_smul (f : Lp 𝕜 p μ) (g : Lp E q μ) :
```

Comment:
Let's make `lpSMul` a single token
```suggestion
lemma coe_lpSMul (f : Lp 𝕜 p μ) (g : Lp E q μ) :
```

Reply:
Done, but I kept `coeFn` because in `MeasureTheory.Lp`, this is how we distinguish between `↑` and `↑↑`.